clang/lib/Format/NumericLiteralCaseFixer.cpp

```diff
@@ -0,0 +1,174 @@
+//===--- NumericLiteralCaseFixer.cpp ----------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file implements NumericLiteralCaseFixer that standardizes character
+/// case within numeric literals.
+///
+//===----------------------------------------------------------------------===//
+
+#include "NumericLiteralCaseFixer.h"
+#include "NumericLiteralInfo.h"
+
+#include "llvm/ADT/StringExtras.h"
+
+#include <algorithm>
+
+namespace clang {
+namespace format {
+
+static bool isNumericLiteralCaseFixerNeeded(const FormatStyle &Style) {
+  // Check if language is supported.
+  switch (Style.Language) {
+  case FormatStyle::LK_C:
+  case FormatStyle::LK_Cpp:
+  case FormatStyle::LK_ObjC:
+  case FormatStyle::LK_CSharp:
+  case FormatStyle::LK_Java:
+  case FormatStyle::LK_JavaScript:
+    break;
+  default:
+    return false;
+  }
+
+  // Check if style options are set.
+  const FormatStyle::NumericLiteralCaseStyle LeaveAllCasesUntouched{};
+  return Style.NumericLiteralCase != LeaveAllCasesUntouched;
```

Comment:
Does this depend on `NLCS_Leave` being `0`? Consider explicit checks, e.g.:
```suggestion
  const auto &Option = Style.NumericLiteralCase;
  const auto Leave = FormatStyle::NLCS_Leave;
  return Option.Prefix != Leave || Option.HexDigit != Leave ||
         Option.ExponentLetter != Leave || Option.Suffix != Leave;
```